packages/world/src/modules/core/implementations/StoreRegistrationSystem.sol

```diff
@@ -81,6 +81,9 @@ contract StoreRegistrationSystem is System, IWorldErrors, LimitedCallContext {
     // Require the table's namespace to exist
     AccessControl.requireExistence(tableId.getNamespaceId());
 
+    // Require the table to exist
+    AccessControl.requireExistence(tableId);
```

Comment:
I wonder if we should put this in `StoreCore.registerStoreHook` and revert with `IStoreErrors.Store_TableNotFound`?

Reply:
I kinda like that, but generally `StoreCore` doesn't have explicit "does this exist" checks - I think you can `get` and `set` records without the table actually being registered.

So for consistency we would have to add it every Store function.

Reply:
`registerTable` does lots of validation, including checking if a table already exists! I think we can safely add these sorts of validations to registration/install steps (i.e. not in the hot path).